fabtests/unit/mr_cache_evict.c

```diff
@@ -43,17 +43,97 @@
 #include "unit_common.h"
 #include "shared.h"
 
+static bool run_cuda_test = false;
+static void *reuse_addr = NULL;
+static char err_buf[512];
+static size_t mr_buf_size = 16384;
+
+#ifdef HAVE_CUDA_RUNTIME_H
+
+#include <cuda_runtime.h>
+#include <dlfcn.h>
+
+static void *cudart_handle;
+
+static cudaError_t (*cuda_malloc_op)(void **devPtr, size_t size);
+static cudaError_t (*cuda_free_op)(void *devPtr);
+
+static int cuda_init(void)
+{
+	/* If CUDA runtime cannot be opened, assume library cannot be found. */
+	cudart_handle = dlopen("libcudart.so", RTLD_NOW);
+	if (!cudart_handle) {
+		sprintf(err_buf, "Unabled to open libcudart.so");
+		return -FI_ENOSYS;
+	}
+
+	cuda_malloc_op =  dlsym(cudart_handle, "cudaMalloc");
+	if (!cuda_malloc_op) {
+		sprintf(err_buf, "Failed to find cudaMalloc");
+		return -FI_ENOSYS;
+	}
+
+	cuda_free_op = dlsym(cudart_handle, "cudaFree");
+	if (!cuda_free_op) {
+		sprintf(err_buf, "Failed to find cudaFree");
+		return -FI_ENOSYS;
+	}
+
+	return FI_SUCCESS;
+}
+
+static void cuda_shut_down(void)
+{
+	dlclose(cudart_handle);
+}
+
+static void cuda_free(void *ptr)
+{
+	cuda_free_op(ptr);
+}
+
+static void *cuda_malloc(void)
+{
+	cudaError_t ret;
+	void *ptr;
+
+	ret = cuda_malloc_op(&ptr, mr_buf_size);
+	if (ret == cudaSuccess)
+		return ptr;
+	return NULL;
+}
+
+#else
+
+static int cuda_init(void)
+{
+	sprintf(err_buf, "CUDA support not enabled");
+	return -FI_ENOSYS;
+}
+
+static void cuda_shut_down(void)
+{
+}
+
+static void cuda_free(void *ptr)
+{
+}
+
+static void *cuda_malloc(void)
+{
+	return NULL;
+}
+
+#endif /* HAVE_CUDA_RUNTIME_H */
```

Comment:
Please move the cude definitions into a separate .h or .c file, so that other fabtests can use them in the future.